automatilib/cola_auth/i_dot_ai_cola_views.py

```diff
@@ -0,0 +1,155 @@
+import logging
+import re
+from abc import abstractmethod
+from urllib.parse import unquote
+
+import requests
+from django.conf import settings
+from django.contrib.auth import authenticate, get_user_model, login, logout
+from django.http import (
+    HttpRequest,
+    HttpResponse,
+    HttpResponseBadRequest,
+    HttpResponseServerError,
+)
+from django.shortcuts import redirect
+from django.urls import reverse
+from django.views import View
+from jose import jwt
+from jose.exceptions import ExpiredSignatureError, JWTClaimsError, JWTError
+
+LOGGER = logging.getLogger(__name__)
+
+User = get_user_model()
+
+
+def get_cola_cognito_user_pool_jwk() -> tuple[dict, str] or None:
+    cola_issuer = f"https://cognito-idp.{settings.AWS_REGION_NAME}.amazonaws.com/{settings.COLA_COGNITO_USER_POOL_ID}"
+    response = requests.get(f"{cola_issuer}/.well-known/jwks.json", timeout=5)
+    if response.status_code != 200:
+        return None
```

Comment:
should we raise an exception here? genuine question - i dont know

Reply:
I'm raising it here, but we can move it into this if we want to?

jwk_response = get_cola_cognito_user_pool_jwk()
if not jwk_response:
    return HttpResponseServerError()